packages/block-library/src/social-link/patterns.js

```diff
@@ -0,0 +1,285 @@
+/**
+ * Internal dependencies
+ */
+import {
+	AmazonIcon,
+	BandcampIcon,
+	BehanceIcon,
+	ChainIcon,
+	CodepenIcon,
+	DeviantArtIcon,
+	DribbbleIcon,
+	DropboxIcon,
+	EtsyIcon,
+	FacebookIcon,
+	FeedIcon,
+	FivehundredpxIcon,
+	FlickrIcon,
+	FoursquareIcon,
+	GoodreadsIcon,
+	GoogleIcon,
+	GitHubIcon,
+	InstagramIcon,
+	LastfmIcon,
+	LinkedInIcon,
+	MailIcon,
+	MastodonIcon,
+	MeetupIcon,
+	MediumIcon,
+	PinterestIcon,
+	PocketIcon,
+	RedditIcon,
+	SkypeIcon,
+	SnapchatIcon,
+	SoundCloudIcon,
+	SpotifyIcon,
+	TumblrIcon,
+	TwitchIcon,
+	TwitterIcon,
+	VimeoIcon,
+	VkIcon,
+	WordPressIcon,
+	YelpIcon,
+	YouTubeIcon,
+} from './icons';
+
+const patterns = [
+	{
+		isDefault: true,
+		name: 'wordpress',
+		attributes: { site: 'wordpress' },
+		title: 'WordPress',
+		icon: WordPressIcon,
+	},
```

Comment:
Unlike the Embed block, the Social Link block was initially conceived so that it can't be "generic", i.e. it has to follow a pattern. Given its purpose, this makes sense.

On the other hand, the way that Block Variations / Patterns are currently defined, if the "base" type is to be supplanted by the patterns there needs to be a default one. I chose _WordPress_ as the least opinionated one (or the least controversial opinion?).

I guess we could instead adapt the Patterns API so that it's possible to supplant the base type without requiring a default, but I'm not sure it's worth adding that compromise. What do you think, @gziolo and @mtias?

Reply:
Side note: gotta love Vim for how magically it was able to transform `social-list.js` into `patterns.js` with just a quick macro. <3